### PR TITLE
[No QA] Fix deploy announce in slack

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
1. Removes success condition to avoid posting multiple messages
2. Resets ENV for multiple runners

cc @roryabraham @AndrewGable 

### Fixed Issues
$ https://github.com/Expensify/App/issues/8199

### Tests
1. Next time we deploy app, verify:
- The Slack announcement includes the correct deploy version
- The success message is only posted once
- If any platform fails to deploy, no success message is posted and we see a failure announcement instead.

- [X] Verify that no errors appear in the JS console

### CODE REVIEW

File: .github/workflows/platformDeploy.yml [hunk 1]
```diff
@@ -374,9 +374,11 @@ jobs:
   postSlackMessageOnSuccess:
     name: Post a Slack message when all platforms deploy successfully
     runs-on: ubuntu-latest
-    if: ${{ success() }}
     needs: [android, desktop, iOS, web]
     steps:
+      - name: Set version
+        run: echo "VERSION=$(npm run print-version --silent)" >> "$GITHUB_ENV"
```
Comment (roryabraham) [inline .github/workflows/platformDeploy.yml:380]: You'll need to run `actions/checkout` before this will work

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: .github/workflows/platformDeploy.yml (hunk 1 above)
Comment (luacmartins) [reply .github/workflows/platformDeploy.yml:384]: Updated!

File: .github/workflows/platformDeploy.yml [hunk 2]
```diff
@@ -374,9 +374,13 @@ jobs:
   postSlackMessageOnSuccess:
     name: Post a Slack message when all platforms deploy successfully
     runs-on: ubuntu-latest
-    if: ${{ success() }}
```
Comment (roryabraham) [inline .github/workflows/platformDeploy.yml:377]: Why did you remove this?

Comment (luacmartins) [reply .github/workflows/platformDeploy.yml:377]: Oh I saw that multiple messages were posted [here](https://expensify.slack.com/archives/C07J32337/p1646957073891919) and thought that it was also a problem because of [this](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds). 

I took another look and haven't seen multiple success messages being posted again, so it doesn't seem to be a problem after all. 

Updating the code to add the conditional back!

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
Monitor [here](https://github.com/Expensify/App/actions/runs/2000976703) to see if this worked

**Comment by luacmartins** [comment]
Didn't seem to work https://expensify.slack.com/archives/C07J32337/p1647557985727479 😭

**Comment by roryabraham** [comment]
Actually it looks like the workflow run I pointed you towards didn't pick up these changes. We'll just need to wait for another deploy then. I can manually CP this to test if you want?

**Comment by roryabraham** [comment]
Manually CP'ing this to test it.
